fsspec/implementations/local.py

```diff
@@ -59,11 +59,16 @@ def rmdir(self, path):
 
     def ls(self, path, detail=False, **kwargs):
         path = self._strip_protocol(path)
-        if detail:
+        isdir = os.path.isdir(path)
+        if detail and isdir:
             with os.scandir(path) as it:
                 return [self.info(f) for f in it]
-        else:
+        elif detail and not isdir:
+            return [self.info(path)]
+        elif not detail and isdir:
             return [posixpath.join(path, f) for f in os.listdir(path)]
+        else:
+            return [path]
```

Comment:
Do we get here if the path doesn't exist at all?

Reply:
Yes, that's the `not detail and not isdir` branch, which is triggered also if the file does not exist. Should this be a `FileNotFoundError`, maybe in front of the statement?

Reply:
Yes, I would say ls(not-a-file) should give FileNotFOundError